shared_model/validators/field_validator.cpp

```diff
@@ -310,6 +310,9 @@ namespace shared_model {
         ReasonsGroupType &reason,
         const interface::types::SignatureRangeType &signatures,
         const crypto::Blob &source) const {
+      if (boost::size(signatures) == 0) {
```

Comment:
Use `boost::empty` instead. `boost::size` might iterate over the collection, whereas `boost::empty` does not